katas/pokerhands/israel/poker.js

```diff
@@ -0,0 +1,433 @@
+
+            if (caracteristicHand[1] < 10) {
+                yy = '0' + caracteristicHand[1].toString();
+            } else {
+                yy = caracteristicHand[1].toString();
+            }
+
+            if (caracteristicHand[2] < 10) {
+                zz = '0' + caracteristicHand[2].toString();
+            } else {
+                zz = caracteristicHand[2].toString();
+            }
+
+            return parseInt(x + yy + zz);
+        }
+        caracteristicHand = this.isFlush(hand);
+        if (caracteristicHand[0]) {
+            const x = '6';
+            let yy = '';
+            let zz = '';
+
+            if (caracteristicHand[1] < 10) {
+                yy = '0' + caracteristicHand[1].toString();
+            } else {
+                yy = caracteristicHand[1].toString();
+            }
+
+            if (caracteristicHand[2] < 10) {
+                zz = '0' + caracteristicHand[2].toString();
+            } else {
+                zz = caracteristicHand[2].toString();
+            }
+
+            return parseInt(x + yy + zz);
+        }
+        caracteristicHand = this.isFull(hand);
+        if (caracteristicHand[0]) {
+            const x = '7';
+            let yy = '';
+            let zz = '';
+
+            if (caracteristicHand[1] < 10) {
+                yy = '0' + caracteristicHand[1].toString();
+            } else {
+                yy = caracteristicHand[1].toString();
+            }
+
+            if (caracteristicHand[2] < 10) {
+                zz = '0' + caracteristicHand[2].toString();
+            } else {
+                zz = caracteristicHand[2].toString();
+            }
+
+            return parseInt(x + yy + zz);
+        }
+        caracteristicHand = this.isSK(hand);
+        if (caracteristicHand[0]) {
+            const x = '8';
+            let yy = '';
+            let zz = '';
+
+            if (caracteristicHand[1] < 10) {
+                yy = '0' + caracteristicHand[1].toString();
+            } else {
+                yy = caracteristicHand[1].toString();
+            }
+
+            if (caracteristicHand[2] < 10) {
+                zz = '0' + caracteristicHand[2].toString();
+            } else {
+                zz = caracteristicHand[2].toString();
+            }
+
+            return parseInt(x + yy + zz);
+        }
+        caracteristicHand = this.isSF(hand);
+        if (caracteristicHand[0]) {
+            const x = '9';
+            let yy = '';
+            let zz = '';
+
+            if (caracteristicHand[1] < 10) {
+                yy = '0' + caracteristicHand[1].toString();
+            } else {
+                yy = caracteristicHand[1].toString();
+            }
+
+            if (caracteristicHand[2] < 10) {
+                zz = '0' + caracteristicHand[2].toString();
+            } else {
+                zz = caracteristicHand[2].toString();
+            }
+
+            return parseInt(x + yy + zz);
+        }
+    }
+
+    //9
+    isSF(hand) {
```

Comment:
isStraightFlush could be a better name, `SF` is a little bit more cryptic and not so easy to understand